src/main/webapp/js/science_portal_form.js

```diff
@@ -110,70 +110,60 @@
-        var curTypename = _selfPortalForm._sessionTypeMap.session_types[i].name
-
-        // This needs to be done by type, and the type need to be
-        // pushed through to the 'then' so the right
-        // values are populated in the image object
-        var curTypeURL = _selfPortalForm.sessionURLs.images + '?type=' + curTypename
-
-        // Do this so each call has it's own curTypeURL
-        // If the Promise is in this same function, the timing of the ajax
-        // calls is such that curTypeURL is the same for all lists so
-        // there's no effective lookup.
-        _getImageDataForType(curTypeURL, curTypename)
-      }
-    }
 
-    function _getImageDataForType(callURL, sessionType ){
-      Promise.resolve(_getAjaxData(callURL))
+    function getFullImageList(){
+      var fullListURL = _selfPortalForm.sessionURLs.images
+      Promise.resolve(_getAjaxData(fullListURL))
         .then(function (imageList) {
 
-          if (_selfPortalForm._ajaxCallCount !== -9) {
-            var imageIDList = new Array()
-            for (var i = 0; i < imageList.length; i++) {
-              imageIDList.push(imageList[i].id)
+          // init the image list structure to simplify the loop below
+          _selfPortalForm._imageData = {}
+          for (var j=0; j<_selfPortalForm._sessionTypeList.length; j++) {
+            _selfPortalForm._imageData[_selfPortalForm._sessionTypeList[j]] = {
+              'imageList': new Array(),
+              "imageIDList": new Array()
             }
+          }
 
-            _selfPortalForm._ajaxCallCount--
-            _selfPortalForm._imageData.push({
-              "sessionType": sessionType,
-              "imageList": imageList,
-              "imageIDList": imageIDList
-            })
-
-            if (_selfPortalForm._ajaxCallCount === 0) {
-              trigger(_selfPortalForm, cadc.web.science.portal.form.events.onLoadFormDataDone)
+          for (var i=0; i<imageList.length; i++) {
+            var curImage = imageList[i]
+            if (isTypeInList(curImage.type)) {
+              // add into the imageList structure
+              _selfPortalForm._imageData[curImage.type].imageList.push(curImage)
+              _selfPortalForm._imageData[curImage.type].imageIDList.push(curImage.id)
             }
+            // else skip is it's not a type currently supported in the UI for
+            // launching sessions.
           }
 
+          trigger(_selfPortalForm, cadc.web.science.portal.form.events.onLoadFormDataDone)
+
         })
         .catch(function (message) {
           trigger(_selfPortalForm, cadc.web.science.portal.form.events.onLoadImageDataError, message)
         })
     }
 
-    // Q: is the full session data being passed in or just the type?
-    //function getImageListForSessionType(sessionData) {
-    //  return getImageListForType(sessionData.name)
-    //}
-
-
     function getImageListForType(sessionType) {
```

Comment:
Is this function still used somewhere?

Reply:
Yes, it's used here: 

https://github.com/opencadc/science-portal/blob/3a24e68334cd3b78419aaa95c6a5d4b491f20d28/src/main/webapp/js/science_portal.js#L591 

but perhaps that doesn't get called anymore?

Reply:
That's good.

Reply:
Thanks, JJ for answering that - it is still used - at the point someone selects a new session type in the launch form, that function will be called to get the list of images to populate the image dropdown.